lib/dialects/postgres/query.js

```diff
@@ -131,6 +132,10 @@ Query.prototype.run = function(sql, parameters) {
       }
     }
 
+    if (self.options.ensureAffectedRows && result.rowCount < 1) {
+      throw new sequelizeErrors.NoAffectedRowsError('No rows affected by: ' + sql);
+    }
```

Comment:
Would it not be simpler to handle this at the Model.update level since all the abstractions are taking care off at that point?